Lombiq.HelpfulLibraries.OrchardCore/Shapes/ShapeExtensions.cs

```diff
@@ -60,6 +61,7 @@ public static T As<T>(this IShape shape)
     /// name="shapeTable"/>, then a new descriptor is added with binding that uses <paramref name="displayAsync"/>.
     /// If <paramref name="type"/> is null or empty, a new random unique name is generated.
     /// </summary>
+    [Obsolete($"This no longer works with the {nameof(DefaultShapeTableManager)}. Use {nameof(HtmlShape)} instead.")]
```

Comment:
Why doesn't this work anymore, isn't this an unintended breaking change that should be reported in the OC issue tracker?

Also, why didn't any of our tests catch this? One of them should.

Reply:
> isn't this an unintended breaking change that should be reported in the OC issue tracker?

It seemed intentional to me. `shapeTable.Descriptors` is now  a `FrozenDictionary`, meaning it's read-only and [this line](https://github.com/Lombiq/Helpful-Libraries/pull/265/files/c5899d3084d7f8103a8666bc3d52f77de288d9bc..5359efa51f40bdc998a7584559c7c2e21a212c12#diff-4e9972ed44cfa1143a8e183b65f2b5679551be32b3ddb3f58624197d841cd6d4R92) throws an exception. What we do here is kind of hacky anyway, adding a custom `IShape` as I did with [`HtmlShape`](https://github.com/Lombiq/Helpful-Libraries/pull/265/files#diff-cea3a22b01c6eaba78ac9a5e3594ff4551c801d8af150e4a0c130629af4f170cR17) is more elegant and less likely to get me ToS'd by the OC folks.

> Also, why didn't any of our tests catch this? One of them should.

We have a custom `IShapeTableManager` implementation `PerTenantShapeTableManager` used by Lombiq.BaseTheme.

Reply:
So this is something that threw, but only when not BaseTheme is used, right? Since `HtmlShape` is only used in `ScriptModuleResourceFilter`, which in turn in our Vue module, I guess this means that the module broke but only with a built-in them (or if otherwise Base Theme is not enabled).

This is quite specific, I wouldn't expect us to guard against it. However, I'm wondering if the OSOCE UI tests should all use Base Theme instead of enabling one of the built-in themes, and have a separate test for Base Theme itself.

Reply:
I don't think that's necessary, if we go down that road, we'll have to randomly enable or disable modules to see which permutation can cause a weird coincidence.

I've updated `PerTenantShapeTableManager` to also use frozen dictionaries. I didn't see any other meaningful change in `DefaultShapeTableManager`.

Reply:
OK!